test/test_catalog_api.py

```diff
@@ -19,68 +19,566 @@
+    ci.category_id = '#Beverages'
+    ci.modifier_list_info = [cimli]
+    ci.tax_ids = ['#SalesTax']
+    ci.variations = [sc, lc]
+
+    c.item_data = ci
+    return c
+
+
+def build_tea():
+    c = CatalogObject()
+    c.type = 'ITEM'
+    c.id = '#Tea'
+    c.present_at_all_locations = True
+
+    stiv = CatalogItemVariation()
+    stiv.item_id = c.id
+    stiv.name = 'Small'
+    stiv.pricing_type = 'FIXED_PRICING'
+    stiv.price_money = Money(150, 'USD')
+    st = CatalogObject()
+    st.type = 'ITEM_VARIATION'
+    st.id = '#SmallTea'
+    st.present_at_all_locations = True
+    st.item_variation_data = stiv
+
+    ltiv = CatalogItemVariation()
+    ltiv.item_id = c.id
+    ltiv.name = 'Large'
+    ltiv.pricing_type = 'FIXED_PRICING'
+    ltiv.price_money = Money(200, 'USD')
+    lt = CatalogObject()
+    lt.type = 'ITEM_VARIATION'
+    lt.id = '#LargeTea'
+    lt.present_at_all_locations = True
+    lt.item_variation_data = ltiv
+
+    cimli = CatalogItemModifierListInfo()
+    cimli.modifier_list_id = '#Milks'
+
+    ci = CatalogItem()
+    ci.name = 'Tea'
+    ci.description = 'Hot leaf juice'
+    ci.abbreviation = 'Te'
+    ci.category_id = '#Beverages'
+    ci.modifier_list_info = [cimli]
+    ci.tax_ids = ['#SalesTax']
+    ci.variations = [st, lt]
+
+    c.item_data = ci
+    return c
+
+
+def build_sales_tax():
+    co = CatalogObject()
+    co.type = 'TAX'
+    co.id = '#SalesTax'
+    co.present_at_all_locations = True
+
+    t = CatalogTax()
+    t.name = 'Sales Tax'
+    t.calculation_phase = 'TAX_SUBTOTAL_PHASE'
+    t.inclusion_type = 'ADDITIVE'
+    t.percentage = '5.0'
+    t.applies_to_custom_amounts = True
+    t.enabled = True
+
+    co.tax_data = t
+    return co
+
+
+class TestCatalogApi(APITestCase):
     """ CatalogApi unit test stubs """
+    objects_by_client_id = dict()
+    test_objects = []
+
+    def build_test_catalog(self):
+        print("build_test_catalog")
+        self.test_objects = [
+            build_beverages(),
+            build_milks(),
+            build_syrups(),
+            build_coffee(),
+            build_tea(),
+            build_sales_tax()
+        ]
+        batch = CatalogObjectBatch(self.test_objects)
+        batches = [batch]
+        req = BatchUpsertCatalogObjectsRequest(str(uuid.uuid4()), batches)
+        res = self.api.batch_upsert_catalog_objects(req)
+        self.assertTrue(res.errors is None)
+        for m in res.id_mappings:
+            self.objects_by_client_id[m.client_object_id] = m.object_id
+
+    def delete_test_catalog(self):
+        print("delete_test_catalog")
+        while True:
+            res = self.api.list_catalog()
+            if res.objects is None:
+                break
+            ids = set()
+            self.assertTrue(res.errors is None)
+            for co in res.objects:
+                ids.add(co.id)
+                if co.item_data and co.item_data.variations:
+                    for v in co.item_data.variations:
```

Comment:
Do we need to iterate over the variations? I think they get listed in the `list_catalog` anyways